io/plugins/eu.esdihumboldt.hale.io.xsd/src/eu/esdihumboldt/hale/io/xsd/reader/XmlSchemaReader.java

```diff
@@ -709,8 +709,8 @@ else if (item instanceof XmlSchemaNotation) {
 					if (!addedBefore) {
 						// place a marker in the map to prevent loading the
 						// location in the call to loadSchema
-						imports.put(location, targetNamespace);
-						loadSchema(location, importedSchema, imports, progress,
+						index.getImports().put(location, targetNamespace);
```

Comment:
You basically replaced the map of import schema locations to namespace that was previously just kept internally.

Maybe you remember our discussion, that for this task, where we need to identify a location for a namespace, we need to inverse association with the problem to decide on which location would be the best / right one to use. For that we said using the first import encountered as the "main" import for the namespace would be a good heuristic.